.github/workflows/develop.yml

```diff
@@ -149,6 +149,10 @@ jobs:
         run: |
           sudo apt update && sudo apt install -y libfuzzer-14-dev
           cargo install cargo-fuzz
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.8'
```

Comment:
@davidhewitt @messense

We are trying to ship the Python script function with PyO3 primarily here. And since statically link to libpython can be complicated, I'm investigating the way to dynamically link to libpython.

So far, it works well and we can see the CI all passed. But this requires a `libpython3.8.so` can be linked in the system.

Thus, I have two questions:

1. How can I make this binary work well with all the libpython >= 3.8?
2. How can I disable the related functionalities if libpython3.so is not found?

Reply:
cc @discord9 any thoughts here?

Reply:
> cc @discord9 any thoughts here?

Maybe we can use `bindgen`  to create a shared library(call it `libpython_finder` maybe?) with the same symbols as limited python abi, and forward those calls to actual `libpython3.*` at runtime with `libloading`, we would also need to export extra functions to check if `libpython` is found, and config pyo3 to use our shared library, the shared library would not need much effect to mantain after we are done developing it as limited api doesn't change that much.